src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -74,10 +74,22 @@ protected override void Shutdown()
             LaunchRequestArguments launchParams,
             RequestContext<object> requestContext)
         {
-            // Set the working directory for the PowerShell runspace to something reasonable
-            // such as the cwd passed in via launch.json. And in case that is null, use the
-            // folder or the script to be executed.
-            string workingDir = launchParams.Cwd ?? Path.GetDirectoryName(launchParams.Program);
+            // Set the working directory for the PowerShell runspace to the cwd passed in via launch.json. 
+            // In case that is null, use the the folder of the script to be executed.  If the resulting 
+            // working dir path is a file path then extract the directory and use that.
+            string workingDir = launchParams.Cwd ?? launchParams.Program;
+            try
+            {
+                if ((File.GetAttributes(workingDir) & FileAttributes.Directory) != FileAttributes.Directory)
+                {
+                    workingDir = Path.GetDirectoryName(workingDir);
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.Write(LogLevel.Error, "cwd path is bad: " + ex.Message);
```

Comment:
How about?
```csharp
catch (Exception ex)
{
    Logger.Write(LogLevel.Error, "cwd path is bad: " + ex.Message);
    workingDir = Environment.CurrentDirectory;
}
```
I thought about doing a `Path.GetDirectoryName(launchParams.Program)` but they could have specified a bad path to a script file as well and that could throw.  In this case, Set-Location gets a valid path and when we try to execute the script - hopefully they'll get a good error in the dbg console.